src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs

```diff
@@ -75,7 +75,7 @@ public HubConnectionContext(ConnectionContext connectionContext, TimeSpan keepAl
 
         public virtual async Task WriteAsync(HubMessage message)
         {
-            await _writeLock.WaitAsync();
+            await _writeLock.WaitAsync(ConnectionAbortedToken);
```

Comment:
If a connection is aborted inside a hub method then an error will be thrown which will trigger a `CompleteMessage` which will hit this line and then error because the connection is aborted. Check whether token has been cancelled and exit the method first?

Reply:
Or should aborting the connection still allow a complete and close message messages to go through?

Reply:
If the connection is aborted we can't send to it, so we should try not to send a `CompletionMessage`

Reply:
Attempting to send a message after cancel is what causes a `OperationCanceledException` when the connected has been aborted.

It is a simple matter to check the cancellation token and exit from `WriteAsync(HubMessage message)` if the connection has been aborted. Thoughts @davidfowl @anurse